### PR TITLE
Add docs and integration tests for Auto-compaction snapshot status API

### PR DESCRIPTION
**Pull request by maytasm**
Add docs and integration tests for Auto-compaction snapshot status API

### Description
This is a follow up PR to https://github.com/apache/druid/pull/10371
This PR adds documentation and integration tests for Auto-compaction snapshot status API of the above referenced PR.

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/operations/api-reference.md [hunk 1]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
 Returns the total size of segments awaiting compaction for the given dataSource. 
 This is only valid for dataSource which has compaction enabled. 
 
+##### GET
+
+* `/druid/coordinator/v1/compaction/status`
+
+Returns the status and statistics from the latest auto compaction run of all dataSource which has/had auto compaction enabled.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:417]:
```suggestion
Returns the status and statistics from the latest auto compaction run of all dataSources which have/had auto compaction enabled.
```

File: docs/operations/api-reference.md [hunk 2]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+The response payload includes a list of `latestStatus` object. Each `latestStatus` represent the status for a dataSource (which has/had auto compaction enabled).
```
Comment (jon-wei) [inline docs/operations/api-reference.md:418]:
```suggestion
The response payload includes a list of `latestStatus` objects. Each `latestStatus` represents the status for a dataSource (which has/had auto compaction enabled). 
```

File: docs/operations/api-reference.md [hunk 3]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `intervalCountAwaitingCompaction`: total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction)
+* `intervalCountCompacted`: total number of intervals of this datasource that is already compacted with the spec set in the auto compaction config.
+* `intervalCountSkipped`: total number of intervals of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.
+
+##### GET
+
+* `/druid/coordinator/v1/compaction/status?dataSource={dataSource}`
+
+Similar to the API `/druid/coordinator/v1/compaction/status` above but filters response to only return information for the {dataSource} given. 
+Note that {dataSource} given must has/had auto compaction enabled.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:437]:
```suggestion
Note that {dataSource} given must have/had auto compaction enabled.
```

File: docs/operations/metrics.md [hunk 4]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`compactTask/availableSlot/count`|Number of available task slot that can be used for auto compaction tasks in the auto compaction run. (this is max slot minus any currently running compaction task)| |Varies.|
+|`segment/waitCompact/bytes`|Total byte of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|
+|`segment/waitCompact/count`|Total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|
+|`interval/waitCompact/count`|Total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|
+|`segment/compacted/bytes`|Total byte of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
+|`segment/compacted/count`|Total number of segments of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:239]:
```suggestion
|`segment/compacted/count`|Total number of segments of this datasource that are already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 5]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
-|`segment/overShadowed/count`|Number of overshadowed segments.||Varies.|
+|`segment/overShadowed/count`|Number of overshadowed segments.| |Varies.|
 |`segment/unavailable/count`|Number of segments (not including replicas) left to load until segments that should be loaded in the cluster are available for queries.|dataSource.|0|
 |`segment/underReplicated/count`|Number of segments (including replicas) left to load until segments that should be loaded in the cluster are available for queries.|tier, dataSource.|0|
 |`tier/historical/count`|Number of available historical nodes in each tier.|tier.|Varies.|
 |`tier/replication/factor`|Configured maximum replication factor in each tier.|tier.|Varies.|
 |`tier/required/capacity`|Total capacity in bytes required in each tier.|tier.|Varies.|
 |`tier/total/capacity`|Total capacity in bytes available in each tier.|tier.|Varies.|
+|`compact/task/count`|Number of task issued in the auto compaction run.| |Varies.|
+|`compactTask/maxSlot/count`|Number of max task slot that can be used for auto compaction tasks in the auto compaction run.| |Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:233]:
```suggestion
|`compactTask/maxSlot/count`|Max number of task slots that can be used for auto compaction tasks in the auto compaction run.| |Varies.|
```

File: docs/operations/metrics.md [hunk 6]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
 |`segment/count`|Number of used segments belonging to a data source. Emitted only for data sources to which at least one used segment belongs.|dataSource.|< max|
-|`segment/overShadowed/count`|Number of overshadowed segments.||Varies.|
+|`segment/overShadowed/count`|Number of overshadowed segments.| |Varies.|
 |`segment/unavailable/count`|Number of segments (not including replicas) left to load until segments that should be loaded in the cluster are available for queries.|dataSource.|0|
 |`segment/underReplicated/count`|Number of segments (including replicas) left to load until segments that should be loaded in the cluster are available for queries.|tier, dataSource.|0|
 |`tier/historical/count`|Number of available historical nodes in each tier.|tier.|Varies.|
 |`tier/replication/factor`|Configured maximum replication factor in each tier.|tier.|Varies.|
 |`tier/required/capacity`|Total capacity in bytes required in each tier.|tier.|Varies.|
 |`tier/total/capacity`|Total capacity in bytes available in each tier.|tier.|Varies.|
+|`compact/task/count`|Number of task issued in the auto compaction run.| |Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:232]:
```suggestion
|`compact/task/count`|Number of tasks issued in the auto compaction run.| |Varies.|
```

File: docs/operations/metrics.md [hunk 7]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`compactTask/maxSlot/count`|Number of max task slot that can be used for auto compaction tasks in the auto compaction run.| |Varies.|
+|`compactTask/availableSlot/count`|Number of available task slot that can be used for auto compaction tasks in the auto compaction run. (this is max slot minus any currently running compaction task)| |Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:234]:
```suggestion
|`compactTask/availableSlot/count`|Number of available task slots that can be used for auto compaction tasks in the auto compaction run. (this is max slot minus any currently running compaction task)| |Varies.|
```

File: docs/operations/metrics.md [hunk 8]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`segment/waitCompact/bytes`|Total byte of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:235]:
```suggestion
|`segment/waitCompact/bytes`|Total bytes of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction).|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 9]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`segment/waitCompact/count`|Total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:236]:
```suggestion
|`segment/waitCompact/count`|Total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction).|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 10]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`interval/waitCompact/count`|Total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction).|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:237]:
```suggestion
|`interval/waitCompact/count`|Total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction).|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 11]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`segment/compacted/bytes`|Total byte of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:238]:
```suggestion
|`segment/compacted/bytes`|Total bytes of this datasource that are already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 12]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`segment/compacted/count`|Total number of segments of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
+|`interval/compacted/count`|Total number of intervals of this datasource that is already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:240]:
```suggestion
|`interval/compacted/count`|Total number of intervals of this datasource that are already compacted with the spec set in the auto compaction config.|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 13]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`segment/skipCompact/bytes`|Total byte of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:241]:
```suggestion
|`segment/skipCompact/bytes`|Total bytes of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 14]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`segment/skipCompact/count`|Total number of segments of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:242]:
```suggestion
|`segment/skipCompact/count`|Total number of segments of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```

File: docs/operations/metrics.md [hunk 15]
```diff
@@ -222,13 +222,25 @@ These metrics are for the Druid Coordinator and are reset each time the Coordina
...
+|`interval/skipCompact/count`|Total number of intervals of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```
Comment (jon-wei) [inline docs/operations/metrics.md:243]:
```suggestion
|`interval/skipCompact/count`|Total number of intervals of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.|datasource.|Varies.|
```

File: docs/operations/api-reference.md [hunk 16]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+##### GET
+
+* `/druid/coordinator/v1/compaction/status`
+
+Returns the status and statistics from the latest auto compaction run of all dataSource which has/had auto compaction enabled.
+The response payload includes a list of `latestStatus` object. Each `latestStatus` represent the status for a dataSource (which has/had auto compaction enabled). 
+The `latestStatus` object has the following keys:
+* `dataSource`: name of the datasource for this status information
+* `scheduleStatus`: auto compaction scheduling status. Possible values are `NOT_ENABLED` and `RUNNING`. Returns `RUNNING ` if the dataSource has an active auto compaction config submitted otherwise, `NOT_ENABLED`
+* `bytesAwaitingCompaction`: total byte of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction)
```
Comment (jon-wei) [inline docs/operations/api-reference.md:422]:
```suggestion
* `bytesAwaitingCompaction`: total bytes of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction)
```

File: docs/operations/api-reference.md [hunk 17]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `bytesCompacted`: total byte of this datasource that is already compacted with the spec set in the auto compaction config.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:423]:
```suggestion
* `bytesCompacted`: total bytes of this datasource that are already compacted with the spec set in the auto compaction config.
```

File: docs/operations/api-reference.md [hunk 18]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `bytesSkipped`: total byte of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:424]:
```suggestion
* `bytesSkipped`: total bytes of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.
```

File: docs/operations/api-reference.md [hunk 19]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `segmentCountAwaitingCompaction`: total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction)
```
Comment (jon-wei) [inline docs/operations/api-reference.md:425]:
```suggestion
* `segmentCountAwaitingCompaction`: total number of segments of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction)
```

File: docs/operations/api-reference.md [hunk 20]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `segmentCountCompacted`: total number of segments of this datasource that is already compacted with the spec set in the auto compaction config.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:426]:
```suggestion
* `segmentCountCompacted`: total number of segments of this datasource that are already compacted with the spec set in the auto compaction config.
```

File: docs/operations/api-reference.md [hunk 21]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `segmentCountSkipped`: total number of segments of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:427]:
```suggestion
* `segmentCountSkipped`: total number of segments of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.
```

File: docs/operations/api-reference.md [hunk 22]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `intervalCountAwaitingCompaction`: total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that is eligible for auto compaction)
```
Comment (jon-wei) [inline docs/operations/api-reference.md:428]:
```suggestion
* `intervalCountAwaitingCompaction`: total number of intervals of this datasource waiting to be compacted by the auto compaction (only consider intervals/segments that are eligible for auto compaction)
```

File: docs/operations/api-reference.md [hunk 23]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `intervalCountCompacted`: total number of intervals of this datasource that is already compacted with the spec set in the auto compaction config.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:429]:
```suggestion
* `intervalCountCompacted`: total number of intervals of this datasource that are already compacted with the spec set in the auto compaction config.
```

File: docs/operations/api-reference.md [hunk 24]
```diff
@@ -410,6 +410,32 @@ Returns total size and count for each datasource for each interval within given
...
+* `intervalCountSkipped`: total number of intervals of this datasource that is skipped (not eligible for auto compaction) by the auto compaction.
```
Comment (jon-wei) [inline docs/operations/api-reference.md:430]:
```suggestion
* `intervalCountSkipped`: total number of intervals of this datasource that are skipped (not eligible for auto compaction) by the auto compaction.
```

**Review by jon-wei — COMMENTED** [review]
Had some minor comments on the docs but otherwise LGTM

File: docs/operations/api-reference.md (hunk 1 above)
Comment (maytasm) [reply docs/operations/api-reference.md:417]: Done

File: docs/operations/api-reference.md (hunk 2 above)
Comment (maytasm) [reply docs/operations/api-reference.md:418]: Done

File: docs/operations/api-reference.md (hunk 3 above)
Comment (maytasm) [reply docs/operations/api-reference.md:437]: Done

File: docs/operations/metrics.md (hunk 4 above)
Comment (maytasm) [reply docs/operations/metrics.md:239]: Done

File: docs/operations/metrics.md (hunk 5 above)
Comment (maytasm) [reply docs/operations/metrics.md:233]: Done

File: docs/operations/metrics.md (hunk 6 above)
Comment (maytasm) [reply docs/operations/metrics.md:232]: Done

File: docs/operations/metrics.md (hunk 7 above)
Comment (maytasm) [reply docs/operations/metrics.md:234]: Done

File: docs/operations/metrics.md (hunk 8 above)
Comment (maytasm) [reply docs/operations/metrics.md:235]: Done

File: docs/operations/metrics.md (hunk 9 above)
Comment (maytasm) [reply docs/operations/metrics.md:236]: Done

File: docs/operations/metrics.md (hunk 10 above)
Comment (maytasm) [reply docs/operations/metrics.md:237]: Done

File: docs/operations/metrics.md (hunk 11 above)
Comment (maytasm) [reply docs/operations/metrics.md:238]: Done

File: docs/operations/metrics.md (hunk 12 above)
Comment (maytasm) [reply docs/operations/metrics.md:240]: Done

File: docs/operations/metrics.md (hunk 13 above)
Comment (maytasm) [reply docs/operations/metrics.md:241]: Done

File: docs/operations/metrics.md (hunk 14 above)
Comment (maytasm) [reply docs/operations/metrics.md:242]: Done

File: docs/operations/metrics.md (hunk 15 above)
Comment (maytasm) [reply docs/operations/metrics.md:243]: Done

File: docs/operations/api-reference.md (hunk 16 above)
Comment (maytasm) [reply docs/operations/api-reference.md:422]: Done

File: docs/operations/api-reference.md (hunk 17 above)
Comment (maytasm) [reply docs/operations/api-reference.md:423]: Done

File: docs/operations/api-reference.md (hunk 18 above)
Comment (maytasm) [reply docs/operations/api-reference.md:424]: Done

File: docs/operations/api-reference.md (hunk 19 above)
Comment (maytasm) [reply docs/operations/api-reference.md:425]: Done

File: docs/operations/api-reference.md (hunk 20 above)
Comment (maytasm) [reply docs/operations/api-reference.md:426]: Done

File: docs/operations/api-reference.md (hunk 21 above)
Comment (maytasm) [reply docs/operations/api-reference.md:427]: Done

File: docs/operations/api-reference.md (hunk 22 above)
Comment (maytasm) [reply docs/operations/api-reference.md:428]: Done

File: docs/operations/api-reference.md (hunk 23 above)
Comment (maytasm) [reply docs/operations/api-reference.md:429]: Done

File: docs/operations/api-reference.md (hunk 24 above)
Comment (maytasm) [reply docs/operations/api-reference.md:430]: Done

**Review by jon-wei — APPROVED** [review]

**Pull request merged**